### PR TITLE
Organize codes and prepare for a PyPI release

### PR DESCRIPTION
**Pull request by perlinm**
TLDR: codes and configurations rearranged so as to allow installing `qfi_opt` as a python package.

Most files have been moved to a `qfi_opt` directory, and package configurations have been moved to `pyproject.toml`, which is trying to become the "standard" way of specifying python project metadata (see [PEP-518](https://peps.python.org/pep-0518/) and [PEP-621](https://peps.python.org/pep-0621/)).  All developer tool package options (such as line length) have also been moved from `setup.cfg` to `pyproject.toml`.

The recommended way for developers to use the codes in this repository is now to run
```bash
pip install -e '.[dev]'
```
in the root directory of the repo, and then to use `qfi_opt` as a package, e.g.
```python
from qfi_opt import spin_models
from qfi_opt.examples.calculate_qfi import compute_QFI
```
Coming soon: installing this package anywhere with `pip install qfi_opt`.

Apologies in advance if merging your local codes with the changes in this PR is a pain!  I promise that these changes will ultimately make using QFI-Opt less painful in the long run.  Example: we no longer need to use hacks like
```python
import sys
sys.path.append("../")
import spin_models
```
in sub-directories of `qfi_opt` (such as [here](https://github.com/jmlarson1/QFI-Opt/blob/869491e2b41f0ddef2c20c56c6581ee4f4fc6342/plotting_scripts/plot_kink.py#L2-L9)), since now we can simply do `from qfi_opt import spin_models`.

I encourage you to organize your files into directories now as you add them (since it's now painless to have files call each other for importing)!

### CODE REVIEW

**Review by lewisswan1 — APPROVED** [review]

**Review by lewisswan1 — APPROVED** [review]

**Comment by perlinm** [comment]
[Will merge once @sriharikrishna gives the okay]
